cachix/src/Cachix/Client/Commands.hs

```diff
@@ -224,11 +227,22 @@ getPushParams env pushOpts name = do
   pushSecret <- findPushSecret (config env) name
   store <- wait (storeAsync env)
   authToken <- Config.getAuthTokenMaybe (config env)
+  compressionMode <- case authToken of
```

Comment:
This should rather case on `PushSecret` and disable this mode for when signing key is used.

Reply:
What the difference between auth token and push token in my case? AFAIU, I should just check PushSecret constructor and ignore its fields, right?

Reply:
Almost zero, but you can have auth token and signing key, so it's best to look into `PushSecret` to make sure we cover both cases correctly. No need to look at the fields.